theano/gpuarray/elemwise.py

```diff
@@ -41,6 +41,28 @@ def get_scal(dt):
     return scalar.get_scalar_type(dt)
 
 
+def max_inputs_to_GpuElemwise(node_or_outputs):
+    if isinstance(node_or_outputs, Apply):
+        outputs = node_or_outputs.outputs
+    else:
+        outputs = node_or_outputs
+    ptr_size = 8
+    # We compile code that would do indexing in int64
+    int_size = 8
+
+    # we take the limit from CUDA for now
+    argument_limit = 232
```

Comment:
If this is the limit, it means we cannot add 2 7D tensors in the general case, which is what happens in the failing case.